gatsby-node.js

```diff
@@ -112,6 +112,10 @@ exports.sourceNodes = ({ actions, createNodeId, createContentDigest }) => {
 	const today = moment().hour(0).minute(0).second(0);
 	for (const rawEvent of events) {
 		promises.push(actions.createNode({
+			// if conferenceLink does not exist in any of the rawEvents
+			// then its value will default to empty, otherwise it will be
+			// overwritten by the actual value
+			conferenceLink: '',
```

Comment:
Why do we need this? Why can't this property be optional so that the page could decide to e.g., make the "join" button unclickable if conferenceLink is undefined?

Reply:
The graphql queries fail and throw an exception if any field is undefined or null. I think there is a way to define the HackEvent type such that a field can be nullable, but thought this was simpler.

Reply:
I haven't tried it for this query, but I have come across a situation where as long as there is one node where `conferenceLink` is actually defined, graphql won't throw an error.

Reply:
Yes @khxia that is true. Adding a conferenceLink to one of the objects makes it a defined field in the HackEvent schema. Right now we just don't have any conference links ready for our events. In the future, there maybe also be times at the start of quarters where there are no links ready but we want to post events for marketing.

Reply:
In that case, could you add a comment with a link to https://github.com/gatsbyjs/gatsby/issues/6800, in order to provide context for what that line is for (and when we can potentially remove it)